fil_actors_shared/src/v9/util/mod.rs

```diff
@@ -16,3 +19,79 @@ mod mapmap;
 mod multimap;
 mod set;
 mod set_multimap;
+
+/// Converts address key to key, equivilent go code can be found at L54-L61 in
+/// <https://github.com/filecoin-project/go-state-types/blob/master/builtin/v9/migration/datacap.go#L54>
+pub fn hamt_addr_key_to_key(addr_key: BytesKey) -> anyhow::Result<BytesKey> {
+    let addr = Address::from_bytes(&addr_key)?;
+    Ok(addr.payload_bytes().into())
+}
```

Comment:
Does this do anything other than error reporting?

Reply:
It removes the network header